src/Swaps/handlers/BaseHandler.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity ^0.8.0;
+
+import {ISwapper} from "../ISwapper.sol";
+import {IEVault, IERC20} from "evk/EVault/IEVault.sol";
+import {SafeERC20Lib} from "evk/EVault/shared/lib/SafeERC20Lib.sol";
+import {RevertBytes} from "evk/EVault/shared/lib/RevertBytes.sol";
+
+abstract contract BaseHandler is ISwapper {
+    uint256 internal constant MODE_EXACT_IN = 0;
+    uint256 internal constant MODE_EXACT_OUT = 1;
+    uint256 internal constant MODE_TARGET_DEBT = 2;
+    uint256 internal constant MODE_MAX_VALUE = 3;
+
+    error Swapper_UnsupportedMode();
+    error Swapper_TargetDebt();
+    error Swapper_TargetDebtBalance();
+
+    function resolveParams(SwapParams memory params) internal view returns (uint256 amountOut, address receiver) {
+        amountOut = params.amountOut;
+        receiver = params.receiver;
+
+        if (params.mode == MODE_EXACT_IN) return (amountOut, receiver);
```

Comment:
shouldn't we revert here? technically, this function should never be used this way

Reply:
hard to say, it's internal asset type check anyway